packages/ui/src/utils/setUserAgent/setUserAgent.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  setCustomUserAgent,
+  SetCustomUserAgentInput,
+} from '@aws-amplify/core/internals/utils';
+
+import {
+  ACCOUNT_SETTINGS_INPUT_BASE,
+  IN_APP_MESSAGING_INPUT_BASE,
+} from './constants';
+
+// public packages only, exclude internal packages e.g. 'react-core', 'ui'
+type PackageName =
+  | 'angular'
+  | 'react'
+  | 'react-auth'
+  | 'react-liveness'
+  | 'react-native'
+  | 'react-native-auth'
+  | 'react-notifications'
+  | 'react-storage'
+  | 'vue';
+
+type ComponentName =
+  | 'AccountSettings'
+  | 'Authenticator'
+  | 'InAppMessaging'
+  | 'FaceLivenessDetector'
+  | 'LocationSearch'
+  | 'MapView';
+
+// semver notation
+type Version = `${string}.${string}.${string}`;
+
+interface SetUserAgentOptions {
+  componentName: ComponentName;
+  packageName: PackageName;
+  version: Version;
+}
+
+/**
+ * @example
+ * ```ts
+ * // set user agent options
+ * const clear = setUserAgent(input);
+ *
+ * // clear user agent options
+ * clear();
+ * ```
+ */
+export const setUserAgent = ({
+  componentName,
+  packageName,
+  version,
+}: SetUserAgentOptions): (() => void) => {
+  let input: SetCustomUserAgentInput | undefined;
+
+  const additionalDetails: SetCustomUserAgentInput['additionalDetails'] = [
+    [componentName],
+    [`ui-${packageName}/${version}`],
```

Comment:
```suggestion
    [`ui-${packageName}`, version],
```

I think you can just do this here, JS will handle formatting the tuple